telemetry/specs/INT.mdk

```diff
@@ -1409,57 +1397,158 @@ INT-MX Header:
-    of the metadata sent in the Telemetry Report [^telem-report], or
-
-  - Send the metadata corresponding to *Instruction Bitmap* and pad the node's INT Metadata
-    with the special all-ones reserved value for a Domain Specific Metadata length,
-    calculated by subtracting from the *Hop ML* a length computed from all bits in the
-    16-bit *Instruction Bitmap*, or
-
-  - Skip INT processing altogether and not send any metadata the monitoring systems.
+* **Optional Domain Specific 'Source-Only' Metadata:** The metadata
+  corresponding to 'source-only' *DS Instruction* bits follows the
+  *DS Instruction* and the *DS Flags* fields in the INT-MX header. The length
+  of this field must be counted in the INT length field of the INT shim header.
 
 * Each INT Transit node along the path that supports INT-MX sends its own metadata
-  values as specified in the *Instruction Bitmap* and *DS Instruction*.
-
-  - If a node is unable to provide a metadata value specified in the
-    instruction bitmap because its value is not available, it must either:
-
-      - ensure that the corresponding bit in *RepMdbits* and/or *DSMdbits* is not set
-        in the Telemetry Report [^telem-report], or
+  values as specified in the *Instruction Bitmap* and *DS Instruction*. Either
+  one of the following two procedures may be implemented:
+
+  1. Modify *RepMdBits* and *DSMdbits* in the Telemetry Report [^telem-report],
+     after copying from INT-MX *Instruction Bitmap* and *DS Instruction*
+     respectively, and recompute *MD Length*.
+
+     - If INT-MX *Instruction Bitmap* bit 0 was set, clear *RepMdBits* bit 0.
+
+     - If a node is unable to provide a metadata value specified in the
+       instruction bitmap because its value is not available, or because it
+       corresponds to a reserved bit, the node must ensure that the
+       corresponding bit in *RepMdbits* and/or *DSMdbits* is not set in the
+       Telemetry Report sent to the monitoring system.
+
+     - If the *Domain Specific ID* does not match any Domain ID known to this
+       node, then the transit node is required to either:
+       - Send the metadata corresponding to *Instruction Bitmap* and ensure that
+         *DSMdbits* is not set in the Telemetry Report and the *MD Length* field
+         reflects the length of the metadata sent in the Telemetry Report, or
+       - Skip INT processing altogether and not send any metadata to the
+         monitoring systems.
+
+     - The INT node should make every effort to include 'source-only' metadata
+       (that was received in the INT-MX header) in the Telemetry Report. There
+       are several ways to accomplish this:
+       - The truncated packet in the *Individual Report Inner Contents* includes
+         the received INT-MX header with embedded 'source-only' metadata, or
+       - The domain specific extension data includes the 'source-only' metadata,
```

Comment:
The reason I did not do that is because there are two ways to report domain specific extension data, *Additional DS Extension Data* in *Individual Report Inner Contents*, or *Domain Specific Extension Data* in TLV in *Individual Report Inner Contents*. Similar thing for the truncated packet.

Reply:
I see. It's a bit hard to make sense out of the free form "domain specific extension data" and the two different ways of doing it by just reading the TR spec.
 
@rsivakolundu 
do you need this 2nd option of using the "domain specific extension data" to carry MX source-only metadata? If not, I suggest we drop this option here to minimize confusion. 
No matter what we mention this option here or not, it's implicitly allowed anyway because the domain specific extension data is a complete free-form.

Reply:
I numbered the 3 options and added to the next paragraph:

> The expectation is that each INT node implementation will only support a subset, while monitoring systems should support at least 1 and 3.

Reply:
and I added to 2:

> 2. The domain specific extension data includes the 'source-only' metadata (note that this is free format, so this may require additional implementation effort by the monitoring system)

Reply:
Why not use the Variable Optional DSMdBits Data to carry "source-only" metadata?

Reply:
@rsivakolundu Are you ok with removing the 2nd option of using "domain specific extension data" to carry MX source-only metadata, leaving the truncated packet and Variable Optional DSMdBits Data options?